sdk/nodejs/runtime/invoke.ts

```diff
@@ -226,8 +226,10 @@ function invokeSync(tok: string, props: any, opts: InvokeOptions, syncInvokes: S
         }
 
         if (provider.__registrationId === undefined) {
-            log.warn(
-`Synchronous call made to "${tok}" with an unregistered provider.
+            // Have to do an explicit console.log here as the call to utils.promiseResult may hang
+            // node, and that may prevent our normal logging calls from making it back to the user.
+            console.log(
+`Synchronous call made to "${tok}" with an unregistered provider. This may cause the program to hang.
 For more details see: https://www.pulumi.com/docs/troubleshooting/#synchronous-call`);
```

Comment:
i've added the "this may cause the program to hang bit" to make it very clear if your'e encountering the hang that this message you're receiving is the path to fixing things.